ToolChain/Assembler/Parser/Parser.py

```diff
@@ -20,586 +19,393 @@
+        This state contains all possible instructions for that particular form.
+        :param form: str, The form of the instruction based on its arguments (ex: instruction-register-immediate)
+        :return: str, The appropriate state string for the instruction form
         """
 
-        instructionCode = None
-        instructionForm = None
+        try:
+            state = STATE_LIST[form]
 
-        for code in possibleCodes:
-            for form in formDescription:
-                if formDescription[form]["typeCode"] is not (code >> 4):
-                    continue
-                found = True
-                for subElem in formDescription[form]["description"]:
-                    if subElem == "instructionCode":
-                        continue
-                    if getattr(partialInstruction, subElem) is None:
-                        # This is not what we are looking for
-                        found = False
-                        break
-                    else:
-                        found = True
-
-                if found:
-                    instructionCode = code
-                    instructionForm = form
-
-        goodInstructionParams = ["instructionCode",
-                                 "sourceRegister",
-                                 "destinationRegister",
-                                 "sourceImmediate",
-                                 "destinationImmediate",
-                                 "width",
-                                 "flags",
-                                 "instructionLength"]
-
-        # This deals with unnecessarily long instructions. Ex: JMP $A $B <- one too many register
-        for param in goodInstructionParams:
-            if param not in formDescription[instructionForm]["description"]:
-                insParam = getattr(partialInstruction, param)
-                if insParam is not None:
-                    raise ValueError("Invalid instruction provided")
-
-        return instructionCode, instructionForm
-
-    def _generateBinaryInstructionFromInstruction(self, instruction: Instruction=None, form: dict=None):
+        except KeyError as e:
+            # Form is invalid, doesn't fit any state descriptions
+            raise ValueError("Invalid instruction format at line " + str(self.lineno))
+
+        return state
+
+    def _evaluateIndicatorData(self, text, dataIdentifier, line, instruction, labelFlag):
         """
-        This is in charge of generating the actual binary code for an instruction. It will take each parts
-        of the instruction and build a binary number with it by using AND logic and bit shifts
-        :param instruction: Instruction, the instruction to work with
-        :param form: dict, a form dict that allows creation of the code
-        :return:
+        This method indicates that we are dealing with identifiers, meaning either dataAlpha, dataNumeric, .global,
```

Comment:
Should be .dataAlpha and .dataNumeric.
Also, the method does not indicates. I would rephrase the comment.